### ISSUE TITLE
Should we use a fixed MemTotal value instead of dynamic one in meminfo?

### ISSUE THREAD

**Issue by YvesChan** [google/gvisor#104]
The default value of MemTotal in /proc/meminfo is 2G, and it'll scale up when the processes is using up all the memory. But in some situations when the process initialize with the current meminfo data, and it cannot sense the change in memory.
So why not using a fixed value passed in by the host(--memory or MemTotal in host as default) instead of dynamic value?

**Comment by prattmic** [comment]
There is no perfect solution. gVisor does not have a fixed memory limit, it can scale just like a normal process. We could report the maximum memory on the system (or the memcg limit, if applicable), but in the past we've found some applications that will use extra memory when they see more available from the host, so reporting a lower number can be advantageous to keep usage lower. This is basically what you describe, but detecting more memory is not always positive.

**Comment by YvesChan** [comment]
> We could report the maximum memory on the system (or the memcg limit, if applicable)
Is it possible for the latest version of gVisor? I think the --memory option is ignored when runsc bootup.

> detecting more memory is not always positive.
Yes but I raised this issue not only for performance purpose. Actually our Java application(Tomcat) could not bootup because of the Memtotal did NOT scale up automatically while loading tons of jar, and it failed with OOM. So we thought in such circumstance, maybe it's better to **provide the ability** for user to setup their own Memtotal value.

**Comment by fvoznika** [comment]
> Is it possible for the latest version of gVisor? I think the --memory option is ignored when runsc bootup.
Correct, the --memory option is ignored right now, but we'd like to support it. Setting `/proc/meminfo:MemTotal` to `--memory` value is a good idea. A patch with the change is highly appreciated.

**Comment by prattmic** [comment]
I agree, we should provide more control over this. I simply meant the explain the reasoning for the existing behavior above.
